File: app/src/main/java/com/bugsnag/android/OnSendCallback.java
package com.bugsnag.android;

import androidx.annotation.NonNull;

/**
 * A callback to be invoked before an {@link Event} is uploaded to a server. Similar to
 * {@link OnErrorCallback}, an {@code OnSendCallback} may modify the {@code Event}
 * contents or even reject the entire payload by returning {@code false}.
 */
public interface OnSendCallback {
    boolean onSend(@NonNull Event event);
}

File: app/src/main/java/com/bugsnag/android/OnSendCallback.kt
package com.bugsnag.android

/**
 * A callback to be invoked before an [Event] is uploaded to a server. Similar to
 * [OnErrorCallback], an `OnSendCallback` may modify the `Event`
 * contents or even reject the entire payload by returning `false`.
 */
fun interface OnSendCallback {
    fun onSend(event: Event): Boolean
}
